tests/unit/models/test_filter_models.py

```diff
@@ -0,0 +1,187 @@
+
+
+@pytest.mark.parametrize(
+    "false_format_datetime", ["2022/12/12 12-12-12", "notadate", 1]
+)
+def test_filter_model_datetime_with_wrong_format_fails(
+    false_format_datetime: str,
+):
+    """Test that the flavor base model fails with long names."""
+    with pytest.raises(ValueError):
+        TestFilterModel(datetime_field=false_format_datetime)
+    for filter_op in GenericFilterOps.values():
+        with pytest.raises(ValueError):
+            TestFilterModel(
+                datetime_field=f"{filter_op}:{false_format_datetime}"
+            )
+
+
+def test_filter_model_int_fields_accept_correct_filter_ops():
+    """Test that the flavor base model fails with long names."""
+    for filter_op in GenericFilterOps.values():
+        if filter_op in NumericFilter.ALLOWED_OPS:
+            int_value = 3
+            model_instance = TestFilterModel(
+                int_field=f"{filter_op}:{int_value}"
+            )
+            assert model_instance.list_of_filters[0]
+            assert model_instance.list_of_filters[0].operation == filter_op
+            assert model_instance.list_of_filters[0].value == int_value
+            assert model_instance.list_of_filters[0].column == "int_field"
+        else:
+            # Fail if wrong filter ops are used (e.g. contains)
+            int_value = 3
+            with pytest.raises(ValueError):
+                TestFilterModel(int_field=f"{filter_op}:{int_value}")
+
+
+def test_filter_model_uuid_fields_accept_correct_filter_ops():
+    """Test that the flavor base model fails with long names."""
+
+    for filter_op in GenericFilterOps.values():
+        if filter_op in UUIDFilter.ALLOWED_OPS:
+            if filter_op == GenericFilterOps.EQUALS:
+                uuid_value = uuid.uuid4()
+                model_instance = TestFilterModel(
+                    uuid_field=f"{filter_op}:{uuid_value}"
+                )
+            else:
+                uuid_value = "a92k34"
+                model_instance = TestFilterModel(
+                    uuid_field=f"{filter_op}:{uuid_value}"
+                )
+            assert model_instance.list_of_filters[0]
+            assert model_instance.list_of_filters[0].operation == filter_op
+            assert model_instance.list_of_filters[0].value == uuid_value
+            assert model_instance.list_of_filters[0].column == "uuid_field"
+        else:
+            # Fail if wrong filter_ops are used (e.g. LTE)
+            with pytest.raises(ValueError):
+                uuid_value = "a92k34"
+                TestFilterModel(uuid_field=f"{filter_op}:{uuid_value}")
+
+
+def test_filter_model_uuid_fields_equality_with_invalid_uuid_fails():
+    """Test filtering for equality with invalid UUID fails."""
+
+    with pytest.raises(ValueError):
+        uuid_value = "a92k34"
+        TestFilterModel(uuid_field=f"{GenericFilterOps.EQUALS}:{uuid_value}")
+
+
+def test_filter_model_string_fields_accept_correct_filter_ops():
```

Comment:
a lot of those tests are doing the same logic, perhaps you could abstract all this into a util `_test_filter_model_accept_correct_filter_ops(filter: Filter, filter_model: FilterBaseModel, filter_field: str, filter_value: Any)` that all these tests then just call with different args. That would make it much easier to write new tests for potential new filters in the future.

Reply:
also why do they all need to define custom filter models actually, can't they just use `TestFilterModel` for the successful checks as well?

Reply:
Abstracted filter model test logic into helper method `_test_filter_model` as outlined above